Panic due to machine id already generated

### Describe the bug

I upgraded to 0.306.0 today and evcc doesn't start any more. The issue can be reverted by downgrading manually with `apt install evcc=0.305.1`

The binary panics because this error message:
```log
panic: machine id already generated
```

### Steps to reproduce

1. Install evcc v. 0.306.0
2. Start evcc
3. evcc doesn't listen on the port


### Configuration details

```yaml
Not related cause 0.305.1 works
```

### Log details

```text
Here are the logs from the systemd service:
May 01 15:04:27 debian systemd[1]: Started evcc.service - evcc.
May 01 15:04:27 debian evcc[4129]: [main  ] INFO 2026/05/01 15:04:27 evcc 0.306.0
May 01 15:04:27 debian evcc[4129]: [main  ] INFO 2026/05/01 15:04:27 using config file: /etc/evcc.yaml
May 01 15:04:27 debian evcc[4129]: [main  ] INFO 2026/05/01 15:04:27 using sqlite database: /var/lib/evcc/evcc.db?_pragma=busy_timeout(5000)&_pragma=foreign_keys(1)&_pragma=syn>
May 01 15:04:27 debian evcc[4129]: panic: machine id already generated
May 01 15:04:27 debian evcc[4129]: goroutine 1 [running]:
May 01 15:04:27 debian evcc[4129]: github.com/evcc-io/evcc/util/machine.CustomID({0x35be48694740?, 0x298f717?})
May 01 15:04:27 debian evcc[4129]:         github.com/evcc-io/evcc/util/machine/machine.go:24 +0xf0
May 01 15:04:27 debian evcc[4129]: github.com/evcc-io/evcc/cmd.configureEnvironment(0x4d8b560, 0x4d8e480)
May 01 15:04:27 debian evcc[4129]:         github.com/evcc-io/evcc/cmd/setup.go:612 +0x225
May 01 15:04:27 debian evcc[4129]: github.com/evcc-io/evcc/cmd.runRoot(0x4d8b560, {0x298c76e?, 0x7?, 0x297f717?})
May 01 15:04:27 debian evcc[4129]:         github.com/evcc-io/evcc/cmd/root.go:190 +0x39b
May 01 15:04:27 debian evcc[4129]: github.com/spf13/cobra.(*Command).execute(0x4d8b560, {0x35be47d3e040, 0x0, 0x0})
May 01 15:04:27 debian evcc[4129]:         github.com/spf13/cobra@v1.10.2/command.go:1019 +0xafb
May 01 15:04:27 debian evcc[4129]: github.com/spf13/cobra.(*Command).ExecuteC(0x4d8b560)
May 01 15:04:27 debian evcc[4129]:         github.com/spf13/cobra@v1.10.2/command.go:1148 +0x465
May 01 15:04:27 debian evcc[4129]: github.com/spf13/cobra.(*Command).Execute(...)
May 01 15:04:27 debian evcc[4129]:         github.com/spf13/cobra@v1.10.2/command.go:1071
May 01 15:04:27 debian evcc[4129]: github.com/evcc-io/evcc/cmd.Execute()
May 01 15:04:27 debian evcc[4129]:         github.com/evcc-io/evcc/cmd/root.go:134 +0x1a
May 01 15:04:27 debian evcc[4129]: main.main()
May 01 15:04:27 debian evcc[4129]:         github.com/evcc-io/evcc/main.go:44 +0x2d
May 01 15:04:27 debian systemd[1]: evcc.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
May 01 15:04:27 debian systemd[1]: evcc.service: Failed with result 'exit-code'.
May 01 15:04:35 debian systemd[1]: Stopped evcc.service - evcc.
```

### What type of operating system or environment does evcc run on?

Linux

### External automation

- [x] I have made sure that no external automation like HomeAssistant or Node-RED is active or accessing any of the mentioned devices when this issue occurs.

### Nightly build

- [x] I have verified that the issue is reproducible with the latest nightly build

### Version

0.306.0

--- a/util/sponsor/auth.go
+++ b/util/sponsor/auth.go
@@ -36,8 +36,11 @@ var (
 	mu                            sync.RWMutex
 	Subject, Token, ActivationKey string
 	ExpiresAt                     time.Time
-	machineID                     = machine.ProtectedID("evcc-sponsor")
 )
+
+func machineID() string {
+	return machine.ProtectedID("evcc-sponsor")
+}
 
 const unavailable = "sponsorship unavailable"
 
@@ -68,7 +71,7 @@ func ActivateSponsorship(licenseKey, email string) (string, error) {
 	res, err := client.Activate(ctx, &pb.ActivateRequest{
 		Key:       licenseKey,
 		Email:     email,
-		MachineId: machineID,
+		MachineId: machineID(),
 	})
 
 	if err != nil {

--- a/util/sponsor/hardware.go
+++ b/util/sponsor/hardware.go
@@ -39,7 +39,7 @@ func checkHardware(vendor string, metadata map[string]string) string {
 
 	client := pb.NewAuthClient(conn)
 	res, err := client.IsAuthorizedHardware(ctx, &pb.HardwareRequest{
-		MachineId: machineID,
+		MachineId: machineID(),
 		Vendor:    vendor,
 		Metadata:  metadata,
 	})

